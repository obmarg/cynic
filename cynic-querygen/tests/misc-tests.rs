use insta::assert_snapshot;

use cynic_querygen::{document_to_fragment_structs, QueryGenOptions};

#[test]
fn mutation_with_scalar_result_and_input() {
    let schema = include_str!("../../schemas/raindancer.graphql");
    let query = include_str!("queries/misc/mutation_with_scalar_result_and_input.graphql");

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn book_subscription_test() {
    let schema = include_str!("../../schemas/books.graphql");
    let query = include_str!("queries/misc/books-subscription.graphql");

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn keyword_renames() {
    let schema = include_str!("../../schemas/test_cases.graphql");
    let query = include_str!("queries/misc/keyword-renames.graphql");

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn test_scalar_casing() {
    let schema = include_str!("../../schemas/test_cases.graphql");
    let query = include_str!("queries/misc/scalar-casing.graphql");

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn test_recursive_inputs() {
    let schema = include_str!("../../schemas/test_cases.graphql");
    let query = r#"
      query MyQuery($input: SelfRecursiveInput!, $input2: RecursiveInputParent!) {
        recursiveInputField(recursive: $input, recursive2: $input2)
      }
    "#;

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn test_string_escaping() {
    let schema = include_str!("../../schemas/test_cases.graphql");
    let query = r#"
      query {
        one: fieldWithStringArg(input: "Hello\n There")
        two: fieldWithStringArg(input: "Hello \" I am a string with quotes")
      }
    "#;

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn test_with_named_schema() {
    let schema = include_str!("../../schemas/github.graphql");
    let query = include_str!("queries/github/query-with-all-derives.graphql");

    assert_snapshot!(document_to_fragment_structs(
        query,
        schema,
        &QueryGenOptions {
            schema_name: Some("my-schema".into()),
            ..QueryGenOptions::default()
        }
    )
    .expect("QueryGen Failed"))
}

#[test]
fn pascal_type_renaming() {
    let schema = r#"
    type my_query {
        field(in: input_type): nested_type
    }

    type nested_type {
        scalar: my_scalar
    }

    input input_type {
        en: an_enum
    }

    enum an_enum {
        value
    }

    schema {
        query: my_query
    }

    scalar my_scalar
    "#;

    let query = r#"
    query my_query($input: input_type) {
        field(in: $input) {
            scalar
        }
    }
    "#;

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}

#[test]
fn test_keyword_arguments() {
    let schema = include_str!("../../schemas/test_cases.graphql");
    let query = include_str!("queries/misc/keyword-argument.graphql");

    assert_snapshot!(
        document_to_fragment_structs(query, schema, &QueryGenOptions::default())
            .expect("QueryGen Failed")
    )
}
