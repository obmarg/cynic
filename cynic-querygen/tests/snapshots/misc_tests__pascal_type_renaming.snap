---
source: cynic-querygen/tests/misc-tests.rs
expression: "document_to_fragment_structs(query, schema,\n        &QueryGenOptions::default()).expect(\"QueryGen Failed\")"
---
#[derive(cynic::QueryVariables, Debug)]
pub struct MyQueryVariables {
    pub input: Option<InputType>,
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(graphql_type = "my_query", variables = "MyQueryVariables")]
pub struct MyQuery {
    #[arguments(in: $input)]
    pub field: Option<NestedType>,
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(graphql_type = "nested_type")]
pub struct NestedType {
    pub scalar: Option<MyScalar>,
}

#[derive(cynic::Enum, Clone, Copy, Debug)]
#[cynic(graphql_type = "an_enum")]
pub enum AnEnum {
    #[cynic(rename = "value")]
    Value,
}

#[derive(cynic::InputObject, Debug)]
#[cynic(graphql_type = "input_type")]
pub struct InputType {
    pub en: Option<AnEnum>,
}

#[derive(cynic::Scalar, Debug, Clone)]
#[cynic(graphql_type = "my_scalar")]
pub struct MyScalar(pub String);

#[cynic::schema]
mod schema {}

